lib/cadet_web/router.ex

```diff
@@ -46,10 +49,20 @@ defmodule CadetWeb.Router do
     get("/devices/:secret/mqtt_endpoint", DevicesController, :get_mqtt_endpoint)
   end
 
-  # Authenticated Pages
+  # Authenticated Pages without course
   scope "/v2", CadetWeb do
     pipe_through([:api, :auth, :ensure_auth])
 
+    get("/user", UserController, :index)
+    get("/user/latest_viewed", UserController, :get_latest_viewed)
+    put("/user/latest_viewed/:course_id", UserController, :update_latest_viewed)
```

Comment:
The `course_id` should not be in the path in this case. You are updating the latest viewed course, not "the latest viewed of course X", whatever that means.

(i.e. just `put("/user/latest_viewed", UserController, :update_latest_viewed)` and have the `course_id` in the body like `{"course_id": 123}`. Note that you won't have to change the controller at all, because Phoenix merges the JSON body into the params, but the frontend will need to be tweaked.)